### PR TITLE
EVG-7194 don't reload data when changing metric

### PR DESCRIPTION
**Pull request by john-m-liu**
Previously we were doing a full load of data from evergreen/cedar when the metric changed. Now we're caching any loads of the data and just filtering the bound data when it changes

### CODE REVIEW

File: public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js [hunk 1]
```diff
@@ -92,6 +92,39 @@ mciModule.controller('PerformanceDiscoveryCtrl', function (
     vm.updateData(true);
   }
 
+  $scope.changeMetric = function () {
+    const fromVersion = vm.fromSelect.selected
```
Comment (jimoleary) [inline public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js]: I don't think any of these constants are being used here.

File: public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js [hunk 2]
```diff
@@ -92,6 +92,39 @@ mciModule.controller('PerformanceDiscoveryCtrl', function (
...
+    const toVersion = vm.toSelect.selected
+    const expandedCurrent = vm.expandedOptions && vm.expandedOptions.includes("current");
+    const expandedBaseline = vm.expandedOptions && vm.expandedOptions.includes("baseline");
+    const expandedHistory = vm.expandedOptions && vm.expandedOptions.includes("history");
+    updateColumnDefs(vm.gridApi.grid);
+
+    vm.isLoading = true;
+    $q.all({
+        fromVersionObj: dataUtil.getCompItemVersion(fromVersion),
```
Comment (jimoleary) [inline public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js]: Are the __dataUtil.getCompItemVersion__ calls needed? loadCompOptions use the returned promise, but this version does not AFAICS.

I think you could replace with the following:

```
      dataUtil.changeMetric(vm.metric_name)
      .then(function (res) {
        vm.gridOptions.data = res
        vm.all_metrics = _.chain(res).map(test => test.all_metrics).flatten().uniq().sort().value();
        // Apply options data to filter drop downs
etc...
```

File: public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js [hunk 3]
```diff
@@ -92,6 +92,39 @@ mciModule.controller('PerformanceDiscoveryCtrl', function (
     vm.updateData(true);
```
Comment (jimoleary) [inline public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js]: it looks like reload is no longer needed, although it's difficult to tell without being able to run the code.

Comment (john-m-liu) [reply public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js]: Reload is still called when you change the data source

File: public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js (hunk 1 above)
Comment (john-m-liu) [reply public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js]: deleted the unused ones

File: public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js (hunk 2 above)
Comment (john-m-liu) [reply public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js]: Good catch, fixed

File: service/templates/perfdiscovery.html [hunk 4]
```diff
@@ -55,7 +55,7 @@
 
     <md-input-container style="width: 180px;">
       <label>Metric</label>
-      <md-select ng-model="$ctrl.metric_name" ng-change="reload()">
+      <md-select ng-model="$ctrl.metric_name" ng-change="changeMetric()">
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply service/templates/perfdiscovery.html]: Yep, see other comment

**Pull request merged**